docs/ml-clustering.md

```diff
@@ -104,4 +104,48 @@ Refer to the [Java API docs](api/java/org/apache/spark/ml/clustering/LDA.html) f
 {% include_example java/org/apache/spark/examples/ml/JavaLDAExample.java %}
 </div>
 
-</div>
\ No newline at end of file
+</div>
+
+## Bisecting k-means
+
+
+Bisecting k-means is a kind of [hierarchical clustering](https://en.wikipedia.org/wiki/Hierarchical_clustering) using a
+divisive (or "top-down") approach: all observations start in one cluster, and splits are performed recursively as one
+moves down the hierarchy.
+
+Bisecting K-means can often be much faster than regular K-means, but it will generally produce a different clustering.
+
+`BisectingKMeans` is implemented as an `Estimator` and generates a `BisectingKMeansModel` as the base model.
+
+The implementation in ML has the following parameters:
+
+* *k*: the desired number of leaf clusters (default: 4). The actual number could be smaller if there are no divisible leaf clusters.
```

Comment:
Your point is valid - this would be a bit out of place in the ml docs. I also agree that is does add a burden of keeping params and defaults in sync with the code. There's a good argument that the param doc lives in the API docs (as it does now for ml). Still, there's also a decent argument for having more detailed docs on params in the user guide, though perhaps only for very important ones (like an initialization scheme, or algorithm type etc).

Indeed, scikit-learn user guide and API docs seem to follow this style (as an example).